codex/sales.nim

```diff
@@ -195,7 +206,8 @@ proc processSlot(
       reprocessSlot = false, returnedCollateral = UInt256.none
   ) {.async.} =
     trace "slot cleanup"
-    await sales.cleanUp(agent, reprocessSlot, returnedCollateral)
+    let fut = sales.cleanUp(agent, reprocessSlot, returnedCollateral)
+    sales.trackedFutures.track(fut)
```

Comment:
I don't think that this is the correct strategy. We're now signaling completion of the worker before we've done cleanup. That means that the worker can pick up the next queue item while the reserved bytes haven't been returned the availability yet.

Maybe it's better to call the `cleanup` function with `noCancel`?